c2corg_ui/static/js/auth/authservice.js

```diff
@@ -89,32 +89,67 @@ app.Authentication.prototype.isAuthenticated = function() {
 /**
  * Checks if the current user has rights to access/edit the document.
  * TODO: rights to personal documents only for current user.
+ * @param {Array<string>} users
+ * @param {?string} imageType
  * @return {boolean}
  * @export
  */
-app.Authentication.prototype.hasEditRights = function(users) {
+app.Authentication.prototype.hasEditRights = function(users, imageType) {
   // not logged -> return false
   if (!this.isAuthenticated()) {
     return false;
   }
-  var userid = this.userData.id;
+
   var roles = this.userData.roles;
   // moderator has rigths
   if (roles.indexOf('moderator') > -1) {
     return true;
   }
-  // we are checking for editing rights of an outing
-  if (users) {
-    for (var i = 0; i < users.length; i++) {
-      if (userid === users[i].id) {
-        return true;
-      }
-    }
-    return false;
-    // by default, user has rights to every doc
+
+  if (users.length) {
```

Comment:
That does not look robust (and explicit for code readers).
For instance "personal articles" should also be editable by associated users only I think.

It would be clearer to check for a type for instance.